src/shipCargo.cpp

```diff
@@ -0,0 +1,76 @@
+#include "shipCargo.h"
+#include "spaceObjects/spaceship.h"
+#include "spaceObjects/playerSpaceship.h"
+#include "gameGlobalInfo.h"
+
+REGISTER_MULTIPLAYER_CLASS(ShipCargo, "ShipCargo");
+
+ShipCargo::ShipCargo() : Cargo("ShipCargo")
+{
+    registerMemberReplication(&callsign);
+    registerMemberReplication(&template_name);
+    registerMemberReplication(&hull_strength);
+}
+
+ShipCargo::ShipCargo(P<ShipTemplate> ship_template) : ShipCargo()
+{
+    template_name = ship_template->getName();
+    callsign = "DRN-" + gameGlobalInfo->getNextShipCallsign();
+    setEnergy(ship_template->energy_storage_amount);
+    hull_strength = ship_template->hull;
+}
+
+ShipCargo::ShipCargo(P<SpaceShip> ship) : ShipCargo()
+{
+    setEnergy(ship->getEnergy());
```

Comment:
```suggestion
    template_name = ship->getTypeName();
    callsign = ship->getCallSign();
    setEnergy(ship->getEnergy());
```

Reply:
Without `template_name` being set there is a crash when selecting the dock after the drone has docked with the carrier

Reply:
right. thanks!